File: langs/javascript/primes/Simple.js
#!/usr/bin/env node

'use strict';

const NUMBER = 1000000;

function getLastPrime(number)
{
    var lastPrime = 2;

    // Traverse each number from 1 to N with the help of for loop
    for (var i = 3; i <= number; i += 2)
    {
        var isPrime = true;
        var limit = Math.sqrt(i);

        for (var j = 3; j <= limit; j += 2) {
            if (i % j == 0) {
                isPrime = false;
                break;
            }
        }

        if (isPrime) {
            lastPrime = i;
        }
    }

    return lastPrime;
}

(async function() {
    const startTimeMs = new Date().getTime();
      
    const lastPrime = getLastPrime(NUMBER);
    console.log(`Last prime: ${lastPrime}`);

    const endTimeMs = new Date().getTime();
    const durationMs = endTimeMs - startTimeMs;
    console.log(`Execution time: ${durationMs}ms`);
})();

File: langs/swift/primes/Simple.swift
#!/usr/bin/swift

import Foundation

let NUMBER = 1000000

func getLastPrime(_ count: Int) -> Int {
    var lastPrime = 2

    for i in stride(from: 3, to: count, by: 2) {
        var isPrime = true
        var limit = Int(sqrt(Double(i)))

        for j in stride(from: 3, to: limit, by: 2) {
            if i % j == 0 {
                isPrime = false
                break
            }
        }

        if isPrime {
            lastPrime = i
        }
    }

    return lastPrime
}

let startTimeMs = Int(Date().timeIntervalSince1970 * 1000)

let lastPrime = getLastPrime(NUMBER)
print("Last prime: \(lastPrime)")

let endTimeMs = Int(Date().timeIntervalSince1970 * 1000)
let durationMs = endTimeMs - startTimeMs

print("Execution time: \(durationMs)ms")
